src/currentprocess/filesource.rs

```diff
@@ -187,81 +132,126 @@ impl StderrSource for super::OSProcess {
     }
 }
 
-// ----------------------- test support for writers ------------------
-
 #[cfg(feature = "test")]
-pub(super) struct TestWriterLock<'a> {
-    inner: MutexGuard<'a, Vec<u8>>,
-}
+mod tests {
```

Comment:
I'd prefer another name, since `tests` is the convention that's used for actual tests. Maybe `test_support` or `test_sources`?